api/instantboxManager.py

```diff
@@ -0,0 +1,106 @@
+import docker
+import random
+import string
+import time
+import json
+
+
+class InstantboxManager(object):
+    CONTAINER_PREFIX = 'instantbox_managed_'
+    TIMEOUT_LABEL = 'org.instantbox.variables.EXPIRATION_TIMESTAMP'
+    OS_LIST = None
+
+    def __init__(self):
+        self.client = docker.from_env()
+
+        try:
+            with open('manifest.json', 'r') as os_manifest:
+                self.OS_LIST = json.load(os_manifest)
+            if self.OS_LIST is None:
+                raise Exception
+        except Exception:
+            raise Exception(
+                'Could not load manifest.json. ' +
+                'Download it from https://get.instantbox.org/manifest.json'
+            )
+
+        self.AVAILABLE_OS_LIST = []
+        for os in self.OS_LIST:
+            for ver in os['subList']:
+                self.AVAILABLE_OS_LIST.append(ver['osCode'])
+
+    def is_create_container(self,
+                            mem,
+                            cpu,
+                            os_name,
+                            os_timeout,
+                            open_port=None):
+
+        if open_port is None:
+            port_dict = {'1588/tcp': None}
```

Comment:
https://docker-py.readthedocs.io/en/stable/containers.html
`None, to assign a random host port. For example, {'2222/tcp': None}.`